pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -205,21 +205,11 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 		return err
 	}
 
-	esUrl, err := url.Parse(ds.URL)
-	if err != nil {
-		logger.Error("Failed to parse data source URL", "error", err, "url", ds.URL)
-		return err
-	}
-
-	resourcePath, err := url.Parse(req.Path)
+	esUrl, err := createElasticsearchURL(req, ds)
 	if err != nil {
-		logger.Error("Failed to parse data source path", "error", err, "url", req.Path)
-		return err
+		logger.Error("Failed to create request url", "error", err, "url", ds.URL, "path", req.Path)
 	}
 
-	// We take the path and the query-string only
-	esUrl.RawQuery = resourcePath.RawQuery
```

Comment:
Hm, are we sure we never need or want to support query parameters in resources calls? Because I think some APIs in ES use GET query parameters, so I think we should also add this to `createElasticsearchURL`. WDYT?

Reply:
Resource path should not have any `RawQuery` based on following check `req.Path != "" && !strings.HasSuffix(req.Path, "/_mapping") && req.Path != "_msearch" && req.Path != "_mapping"`. So it is not needed as there are currently no queries supported. Maybe I can add a comment explaining it - so when someone allows additional path/resource calls, they are aware of needing to implement raw query. 

I also noticed that I haven't add test for why we are doing it - so I will also add test for path with `:` to ensure that it is covered.

But to summarise we can't use `url.Parse(req.Path)` anymore as it throws error for path that includes `:` , so I am worried, that if we wanted to go 1 step ahead and implement raw queries (even if it is not needed), we might introduce some issue related to splitting on `?`. Or do you have any other ideas.

Reply:
https://github.com/grafana/grafana/pull/82327/commits/bbacb5695def59480ad7e49b087d07ed59245545